sparse_strips/vello_hybrid/src/scene.rs

```diff
@@ -368,3 +372,237 @@ impl GlyphRenderer for Scene {
+
+                            if start < adjusted_strips.len() && count > 0 {
+                                let end = (start + count).min(adjusted_strips.len());
+                                let paint = self.encode_current_paint();
+                                self.wide.generate(
+                                    &adjusted_strips[start..end],
+                                    self.fill_rule,
+                                    paint,
+                                    0,
+                                );
+                            }
+
+                            range_index += 1;
+                        }
+                    }
+                    RenderCommand::SetPaint(paint) => {
+                        self.set_paint(paint.clone());
+                    }
+                    RenderCommand::SetPaintTransform(transform) => {
+                        self.set_paint_transform(*transform);
+                    }
+                    RenderCommand::ResetPaintTransform => {
+                        self.reset_paint_transform();
+                    }
+                    RenderCommand::SetTransform(transform) => {
+                        self.set_transform(*transform);
+                    }
+                    RenderCommand::SetFillRule(fill_rule) => {
+                        self.set_fill_rule(*fill_rule);
+                    }
+                    RenderCommand::SetStroke(stroke) => {
+                        self.set_stroke(stroke.clone());
+                    }
+                    RenderCommand::PushLayer {
+                        clip_path,
+                        blend_mode,
+                        opacity,
+                        mask,
+                    } => {
+                        self.push_layer(clip_path.as_ref(), *blend_mode, *opacity, mask.clone());
+                    }
+                    RenderCommand::PopLayer => {
+                        self.pop_layer();
+                    }
+                }
+            }
+        }
+    }
+}
+
+/// Recording management implementation
+impl Scene {
+    /// Prepare cached strips for rendering by adjusting alpha indices and extending alpha buffer
+    #[expect(
+        clippy::cast_possible_truncation,
+        reason = "Alphas length conversion is safe in this case"
+    )]
+    fn prepare_cached_strips(
+        &mut self,
+        cached_strips: &[Strip],
+        cached_alphas: &[u8],
+    ) -> Vec<Strip> {
+        // Calculate offset for alpha indices based on current buffer size
+        let alpha_offset = self.alphas.len() as u32;
+        // Create adjusted strips with corrected alpha indices
+        let mut adjusted_strips = Vec::with_capacity(cached_strips.len());
+        for strip in cached_strips {
+            let mut adjusted_strip = *strip;
+            adjusted_strip.alpha_idx += alpha_offset;
+            adjusted_strips.push(adjusted_strip);
+        }
```

Comment:
Rust will automatically reserve the right capacity if you use an iterator IIUC - that might be preferred to the manual capacity setting here